src/main/java/org/dswarm/graph/batch/Neo4jProcessor.java

```diff
@@ -648,43 +765,8 @@ private Object getProperty(final String key, final Map<String, Object> propertie
 		return Optional.absent();
 	}
 
-	private Optional<Long> getIdFromLongIndex(final long key, final LongLongOpenHashMap tempIndex, final BatchInserterIndex index,
-			final String indexProperty) {
-
-		if (tempIndex.containsKey(key)) {
-
-			return Optional.of(tempIndex.lget());
-		}
+	private boolean checkLongIndex(final long key, final Set<Long> index) {
 
-		if (index == null) {
-
-			return Optional.absent();
-		}
-
-		final IndexHits<Long> hits = index.get(indexProperty, key);
-
-		if (hits != null && hits.hasNext()) {
-
-			final Long hit = hits.next();
-
-			hits.close();
-
-			final Optional<Long> optionalHit = Optional.fromNullable(hit);
-
-			if (optionalHit.isPresent()) {
-
-				// temp cache index hits again
-				tempIndex.put(key, optionalHit.get());
-			}
-
-			return optionalHit;
-		}
-
-		if (hits != null) {
-
-			hits.close();
-		}
-
-		return Optional.absent();
+		return index != null && index.contains(key);
```

Comment:
If I got this right, the index here is one of two field, so I would move the null check to the field initialization and remove it here (as above).